### PR TITLE
chore: update Windows base images to 2023.09B #

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Update Windows base images to 2023.09B.
- WS2019: [September 12, 2023—KB5030214 (OS Build 17763.4851)](https://support.microsoft.com/en-us/topic/september-12-2023-kb5030214-os-build-17763-4851-e6ae7551-49f4-428e-b2d4-caa73078fb06)
- WS2022: [September 12, 2023—KB5030216 (OS Build 20348.1970)](https://support.microsoft.com/en-us/topic/september-12-2023-kb5030216-os-build-20348-1970-34d4aff3-fd05-4270-b288-4ab6379c7f81)

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/windows-image.env [hunk 1]
```diff
@@ -4,18 +4,18 @@
 # CLI example to get the latest image version:
 #   az vm image show --urn MicrosoftWindowsServer:WindowsServer:2019-Datacenter-Core-smalldisk:latest
 WINDOWS_2019_BASE_IMAGE_SKU=2019-Datacenter-Core-smalldisk
-WINDOWS_2019_BASE_IMAGE_VERSION=17763.4737.230802
+WINDOWS_2019_BASE_IMAGE_VERSION=17763.4851.230905
```
Comment (AbelHu) [inline vhdbuilder/packer/windows-image.env:7]: could you help to add links in pr description for these kbs?

**Comment by AbelHu** [comment]
The title is not correct.

File: vhdbuilder/packer/windows-image.env (hunk 1 above)
Comment (junjiezhang1997) [reply vhdbuilder/packer/windows-image.env:7]: Thanks for the reminder. Added it.

**Comment by AbelHu** [comment]
Same issue in the commit message but I think that squash merge should use the pr title so it should not be a problem.

File: vhdbuilder/packer/configure-windows-vhd.ps1 [hunk 2]
```diff
@@ -570,6 +570,37 @@ function Update-Registry {
...
+        }
+        Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name RemoveSourcePortPreservationForRest -Value 1 -Type DWORD
+
+        $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name 2214038156 -ErrorAction Ignore)
+        if (![string]::IsNullOrEmpty($currentValue)) {
+            Write-Log "The current value of 2214038156 is $currentValue"
+        }
+        Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name 2214038156 -Value 1 -Type DWORD
+
+        $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Control\Windows Containers" -ErrorAction Ignore)
```
Comment (AbelHu) [inline vhdbuilder/packer/configure-windows-vhd.ps1:599]: Why do we need to create `HKLM:\SYSTEM\CurrentControlSet\Control\Windows Containers`?

Comment (AbelHu) [reply vhdbuilder/packer/configure-windows-vhd.ps1:599]: Remove it since we do not need to set DeltaHivePolicy    ?

Comment (junjiezhang1997) [reply vhdbuilder/packer/configure-windows-vhd.ps1]: Removed. Also removed "Windows Containers" in write-release-notes-windows.ps1.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**